### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by ronocod**
Cleans up the code as per the Novoda code style and some of the guidelines in *Effective Java*. Hack and Tell coming soon :)

Changes include:
* Getting rid of the `m` and `s` prefixes on instance fields and static fields.
* Encapsulate all of the public mutable fields on `DownloadInfo`
* Remove redundant modifiers (i.e. `final` on a `static` method
* Make fields `final` where possible
* Fix unchecked varargs warning: https://github.com/novoda/download-manager/commit/c6a12bc528102060c74f39c8b71a35a1f5121919
* Specify collection size on creation where possible, e.g. https://github.com/novoda/download-manager/commit/9756009fc6d9b83da626cee84038e0a6ad6648aa
* Add missing support annotations
* Move constructor work to factory methods to allow more mocking, e.g. https://github.com/novoda/download-manager/commit/6d57044644462f6f2282b77dbae4f3d6e2e6d96e

### CODE REVIEW

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java [hunk 1]
```diff
@@ -193,53 +80,50 @@ private Long getLong(String column) {
...
+    private String mediaProviderUri;
+    private int allowedNetworkTypes;
+    private boolean allowRoaming;
+    private boolean allowMetered;
+    private int bypassRecommendedSizeLimit;
+    private long batchId;
 
     /**
      * Result of last {DownloadThread} started by
+<<<<<<< HEAD
```
Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: ru roh

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java [hunk 2]
```diff
@@ -193,53 +80,50 @@ private Long getLong(String column) {
...
+     * {@link #isReadyToDownload(CollatedDownloadInfo)} && {@link #startDownloadIfNotActive(ExecutorService, StorageManager, DownloadNotifier)}.
+=======
      * {@link #isReadyToDownload(DownloadBatch)} && {@link #startDownloadIfNotActive(ExecutorService)}.
+>>>>>>> master
      */
-    private Future<?> mSubmittedTask;
+    private Future<?> submittedThread;
```
Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: calling it a thread is leaking implementation perhaps?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadDeleter.java [hunk 3]
```diff
@@ -25,16 +25,16 @@ public void deleteFileAndDatabaseRow(DownloadInfo info) {
...
 
-        if (!TextUtils.isEmpty(info.mFileName)) {
-            deleteFileIfExists(info.mFileName);
+        if (!TextUtils.isEmpty(info.getFileName())) {
+            deleteFileIfExists(info.getFileName());
             ContentValues blankData = new ContentValues();
             blankData.put(Downloads.Impl._DATA, (String) null);
             resolver.update(info.getAllDownloadsUri(), blankData, null, null);
-            info.mFileName = null;
+            info.setFileName(null);
```
Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadDeleter.java]: unsetFileName() ?

Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadDeleter.java]: :+1: I'd rather keep that for another PR, there's a lot in this one and I wanna keep the field encapsulation as simple as possible

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java (hunk 2 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: I suppose yeah, I didn't think it was a problem here because it's `private` and it has no getter. I'll go with the consensus.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java (hunk 1 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: such sneak https://github.com/novoda/download-manager/commit/8dd57ee7c2f8d941d54af6627c491f1a03d459fc

**Comment by ouchadam** [comment]
:100: adios AOSP

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/2/
Test PASSed.

**Comment by xrigau** [comment]
seems legit :+1: but can't be merged :-1:

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/6/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java [hunk 4]
```diff
@@ -193,53 +80,46 @@ private Long getLong(String column) {
...
-    /**
-     * Add to check if scannable i.e. we don't want internal files to be scanned
-     */
-    public boolean mScannable;
-    public boolean mNoIntegrity;
-    public String mHint;
-    public String mFileName;
-    public String mMimeType;
-    public int mDestination;
-    public int mVisibility;
```
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: What happend to visibility?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java [hunk 5]
```diff
@@ -500,18 +461,18 @@ public boolean startScanIfReady(DownloadScanner scanner) {
     }
 
     public boolean isOnCache() {
-        return (mDestination == Downloads.Impl.DESTINATION_CACHE_PARTITION
-                || mDestination == Downloads.Impl.DESTINATION_SYSTEMCACHE_PARTITION
-                || mDestination == Downloads.Impl.DESTINATION_CACHE_PARTITION_NOROAMING
-                || mDestination == Downloads.Impl.DESTINATION_CACHE_PARTITION_PURGEABLE);
+        return (this.destination == Downloads.Impl.DESTINATION_CACHE_PARTITION
```
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: why `this.` here?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java [hunk 6]
```diff
@@ -907,14 +907,15 @@ private SqlSelection getWhereClause(final Uri uri, final String where, final Str
             selection.appendClause(Downloads.Impl._ID + " = ?", getDownloadIdFromUri(uri));
         }
         if (uriMatch == BATCHES_ID) {
-            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", ContentUris.parseId(uri));
+            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", String.valueOf(ContentUris.parseId(uri)));
```
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: better just use `contentUri.getLastPathSegment()` to get a string.

You might get `null` instead of `"null"`

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java [hunk 7]
```diff
@@ -43,19 +43,22 @@
...
+            case CONNECTIVITY_ACTION:
+                checkConnectivityToStartService(context);
+                break;
+            case ACTION_OPEN:
+            case ACTION_LIST:
+            case ACTION_HIDE:
+            case ACTION_DELETE:
+            case ACTION_CANCEL:
+                handleSystemNotificationAction(context, intent);
+                break;
```
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java]: :+1: looks really readable now!

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java (hunk 4 above)
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: Is this needed for the TODO in https://github.com/novoda/download-manager/blob/8eac8276696c4810e1a81cf6c06d9320c0b15f57/library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java#L674

**Comment by friedger** [comment]
nice work!

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java (hunk 6 above)
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: Afaik `ContentUris.parseId(uri)` returns `-1` in case the path is empty, I don't think there's a case where `null` is involved. 

http://developer.android.com/reference/android/content/ContentUris.html#parseId(android.net.Uri)

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java (hunk 7 above)
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java]: :+1: nice! but no `default` ? Even if the default case is not handled (including just a comment), including one makes it explicit for the reader that we've consciously covered all possible cases and anything else doesn't matter to be handled.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java [hunk 8]
```diff
@@ -84,21 +87,29 @@ public void run() {
...
+                long batchId = getBatchId(intent);
+                hideNotification(context, batchId);
+                break;
+            }
+            case ACTION_CANCEL:
+                cancelBatchThroughDatabaseState(context, intent);
+                break;
+            case ACTION_DELETE:
+                deleteDownloadThroughDatabaseState(context, intent);
+                break;
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java]: :+1: same as https://github.com/novoda/download-manager/pull/63/files?diff=split#r34016507

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java [hunk 9]
```diff
@@ -58,12 +58,16 @@ public void exec(MediaScannerConnection conn) {
         }
     }
 
-    //    @GuardedBy("mConnection")
-    private HashMap<String, ScanRequest> mPending = new HashMap<String, ScanRequest>();
+    //    @GuardedBy("mediaScannerConnection")
+    private HashMap<String, ScanRequest> pendingRequests = new HashMap<>();
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java]: Is there something that requires `HashMap` explicitly? If not, this should be typed against the interface - `Map`.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java [hunk 10]
```diff
@@ -58,12 +58,16 @@ public void exec(MediaScannerConnection conn) {
...
 
-    public DownloadScanner(Context context) {
-        mContext = context;
-        mConnection = new MediaScannerConnection(context, this);
+    public static DownloadScanner newInstance(Context context){
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java]: Have we agreed on using/not using this pattern throughout the project?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java [hunk 11]
```diff
@@ -58,12 +58,16 @@ public void exec(MediaScannerConnection conn) {
...
+        return new DownloadScanner(context.getContentResolver(), context);
+    }
+
+    public DownloadScanner(ContentResolver resolver, Context context) {
+        this.resolver = resolver;
+        mediaScannerConnection = new MediaScannerConnection(context, this);
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java]: This kind of circular dependency, weird.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java (hunk 6 above)
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: @Electryc yes, you are correct for the -1 and for the null as uri matches "batches/#", hence, getLastPathSegement is the way to go. 

Furthermore, appendClause does a "toString" on the arguments. Maybe, we want to change the signature to String  instead of <T>?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java [hunk 12]
```diff
@@ -221,18 +221,18 @@ private void enqueueUpdate() {
...
+        updateHandler.sendMessageDelayed(
+                updateHandler.obtainMessage(MSG_FINAL_UPDATE, lastStartId, -1),
                 5 * MINUTE_IN_MILLIS);
     }
 
     private static final int MSG_UPDATE = 1;
     private static final int MSG_FINAL_UPDATE = 2;
 
-    private Handler.Callback mUpdateCallback = new Handler.Callback() {
+    private final Handler.Callback updateCallback = new Handler.Callback() {
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java]: :+1:

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java (hunk 10 above)
Comment (blundell) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadScanner.java]: yeah this method should just be inlined where-ever it is used

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java [hunk 13]
```diff
@@ -153,7 +153,7 @@ public DownloadBatch retrieveBatchFor(DownloadInfo download) {
 
                 List<DownloadInfo> batchDownloads = new ArrayList<>(1);
                 for (DownloadInfo downloadInfo : downloads) {
-                    if (downloadInfo.batchId == id) {
+                    if (downloadInfo.getBatchId() == id) {
```
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: is batchid still a primitive? if not this should be an equals!

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java [hunk 14]
```diff
@@ -367,17 +367,17 @@ private boolean updateLocked() {
...
-            mAlarmManager.set(AlarmManager.RTC_WAKEUP, now + nextRetryTimeMillis, PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_ONE_SHOT));
+            alarmManager.set(AlarmManager.RTC_WAKEUP, now + nextRetryTimeMillis, PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_ONE_SHOT));
         }
 
         return isActive;
     }
 
     private void updateTotalBytesFor(DownloadInfo info) {
-        if (info.mTotalBytes == -1) {
+        if (info.getTotalBytes() == -1) {
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java]: Maybe using `!info.hasTotalBytes()` or create the opposite method and use that , although I don't really know how to name it (`hasNotTotalBytes` ?).

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java (hunk 13 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: Yep, it's a `long`

File: library/src/main/java/com/novoda/downloadmanager/lib/Helpers.java [hunk 15]
```diff
@@ -400,95 +400,95 @@ private static void parseStatement(Lexer lexer) {
...
+        private int currentToken = TOKEN_START;
 
         public Lexer(String selection, Set<String> allowedColumns) {
-            mSelection = selection;
-            mAllowedColumns = allowedColumns;
-            mChars = new char[mSelection.length()];
-            mSelection.getChars(0, mChars.length, mChars, 0);
+            this.selection = selection;
+            this.allowedColumns = allowedColumns;
+            chars = new char[this.selection.length()];
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/Helpers.java]: `this.chars` for consistency?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java [hunk 16]
```diff
@@ -247,73 +127,154 @@ private Long getLong(String column) {
...
+
+    String getMimeType() {
+        return mimeType;
+    }
+
+    int getDestination() {
+        return destination;
+    }
+
+    int getControl() {
```
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: think these should be public to match the latest convention

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/8/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java [hunk 17]
```diff
@@ -41,52 +42,61 @@
     /**
      * the max amount of space allowed to be taken up by the downloads data dir
      */
-    private static final long sMaxdownloadDataDirSize = 100 * 1024 * 1024;
+    private static final long MAX_DOWNLOAD_DATA_DIR_SIZE = 100 * 1024 * 1024;
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/StorageManager.java]: Worth specifying the unit as well in the name `MAX_DOWNLOAD_DATA_DIR_SIZE_BYTES`?

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java (hunk 4 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java]: Visibility is set on the batch level now: https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/BatchInfo.java#L7-L8

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/9/
Test PASSed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/10/
Test PASSed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/14/
Test PASSed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/15/
Test PASSed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/16/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java (hunk 6 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java]: Changed now: https://github.com/novoda/download-manager/commit/d1472e5d719893d361e1f1631e522d1eb141e8a4
And I've already changed `appendClause()`: https://github.com/novoda/download-manager/pull/63/files#diff-d18483ff140c15dd49ed3d88822df403L233

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/17/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java (hunk 7 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java]: :+1: https://github.com/novoda/download-manager/commit/43f84ea6d77aa05c3177b2998cd9223ae519d614

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java (hunk 8 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java]: :+1: https://github.com/novoda/download-manager/commit/43f84ea6d77aa05c3177b2998cd9223ae519d614

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/19/
Test PASSed.

**Comment by Electryc** [comment]
:100: Nice code cleanup, LGTM!

**Comment by xrigau** [comment]
woooo! :+1:

**Comment by xrigau** [comment]
If we merge #72 before this one then we can see the before and the after :tongue:

**Comment by ronocod** [comment]
@xrigau :100:

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/25/
Test PASSed.

**Comment by ouchadam** [comment]
#72 merged, let's go!
